src/inc_gauged_raz.rs

```diff
@@ -1151,4 +1152,30 @@ mod tests {
 
 		assert_eq!(original_names, new_names);
 	}
+
+
+	#[test]
+	fn test_name_indexes() {
+		use raz_meta::Names;
+		use rand::{thread_rng,Rng};
+
+		let mut r: Raz<_,Names> = Raz::new();
+		for i in 0..1000 {
+			r.push_left(i);
+			if i % 10 == 0 {
+				r.archive_left(::inc_level(),Some(name_of_usize(i)));
+			}	
+		}
+		let t = r.unfocus();
+
+		println!("top: {:?}", t.meta());
+
+		for _ in 0..10 {
+			let val = (thread_rng().gen::<usize>() % 100) * 10;
+			let tree_name = name_pair(name_of_usize(val),name_of_string(String::from("tree")));
```

Comment:
Is there any way to avoid this name pairing and just focus based on `name_of_usize(val)`, to exactly match the name given on line 1166?

I assume that this pairing step is necessary because internally, the naming strategy of the RAZ code does this pairing somewhere to its names.  (Is that correct?)